### ISSUE TITLE
Make it possible to use local stubs

### ISSUE THREAD

**Issue by JukkaL** [JukkaL/mypy#25]
Currently the stubs are only searched automatically from the directory stubs below the directory containing (the translated) mypy.py. It would be useful to also look for stubs in a directory relative to the program being type checked/run.

For example, consider the following hierarchy:

    d/foo.py
    d/stubs/mylib.py

Now if d/foo.py does `import mylib`, it should refer to the mylib.py file in the stubs directory under d. `stubs` is a magic directory name for stubs.

**Comment by JukkaL** [comment]
The .pyi extension seems to address this issue, so closing this now.

**Comment by DionysusG** [comment]
Hi,

This might be a stupid question, but I have tried the both methods to use a local stub: putting the pyi file in the same directory as the py file to run and exporting MYPMPATH to a stub directory and put the pyi file there. I named both pyi file the same name as the py file except for the extension. However, both situations does not throw an error when I run a test program with the incorrect typing hint. 

 Is there some new way to use the new local stub?

**Comment by gvanrossum** [comment]
Can you provide more specific? Show exactly which files are where, what
MYPYPATH is set to, and the error you are getting.

--Guido (mobile)

**Comment by Kamforka** [comment]
So as a follow up I also can't seem to sort out the problems with local stub files.


I have the following project structure:
```
stub-typing/
├── stubtyping/
│   ├── __init__.py
│   ├── main.py
│   └── main.pyi
└── venv/
```

`main.py` has the following content:

```python
def add(a, b):
    return a + b


add(1, 2)
add('a', 'b')
```

and `main.pyi` has:

```python
def add(a: int, b: int) -> int: ...

```

When `MYPYPATH` is unset and  I execute `mypy stubtyping/` I have a blank output instead of  the expected:
```
main.py:6: error: Argument 1 to "add" has incompatible type "str"; expected "int"
main.py:6: error: Argument 2 to "add" has incompatible type "str"; expected "int"
```

I also tried to set`MYPYPATH` to `stubtyping/` but I still get the same results.

What am I missing?

**Comment by gvanrossum** [comment]
Next time please use the Gitter channel (see project README).

The answer is that mypy ignores `main.py` because it finds `main.pyi` in the same directory. This is by design. If you want type annotations, put then in `main.py` and get rid of `main.pyi`.
